### PR TITLE
Add armv6 and armv7 to multiarch CI and fix build issues

### PR DESCRIPTION
**Pull request by ankith26**
This PR mainly does ~~3~~ 2 things
- ~~Fixes arm32 specific warnings across the codebase~~ Has been moved out to be done in a later PR
- Fixes compilation on arm32 (both armv6 and armv7) under the meson build system. As a result of the meson PR, the NEON optimizations are enabled by default on armv7 now (in the legacy buildconfig it is controlled via an opt in flag). Now that it is much easier to pass per file flags, the flags needed to build with `sse2neon.h` are only passed to the files that need it. Therefore, I made it so that only the files that require the `sse2neon.h` header include it. If any file that shouldn't be including it tries to include it, the compilation fails.
- And now coming to the good stuff, we finally have testing on CI to verify that we support both armv6 and armv7 based raspberrypi-like systems! The first level of testing tests that a wheel can be built and unit test can be run on armv6 and armv7. The second level of testing tests that an armv7 wheel can simply be renamed to lie that it's an armv6 wheel, and still pass tests on armv6 hardware. This is important because piwheels does exactly this to package `pygame-ce` wheels over at https://www.piwheels.org/project/pygame-ce/

### CODE REVIEW

File: .github/workflows/build-debian-multiarch.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
         install: |
           apt-get update --fix-missing
           apt-get upgrade -y
-          apt-get install libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev libfreetype6-dev libportmidi-dev libjpeg-dev fontconfig -y
+          apt-get install build-essential libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev libfreetype6-dev libportmidi-dev fontconfig -y
           apt-get install python3-setuptools python3-dev python3-pip python3-wheel python3-sphinx -y
```
Comment (cclauss) [inline .github/workflows/build-debian-multiarch.yml:81]: A suggestion for readability...
```suggestion
          apt-get install build-essential libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev \
                          libsdl2-ttf-dev libfreetype6-dev libportmidi-dev fontconfig python3-dev \
                         python3-pip python3-setuptools python3-sphinx python3-wheel -y
```

Comment (ankith26) [reply .github/workflows/build-debian-multiarch.yml:81]: Good suggestion, but I vaguely remember trying something like this in the past and it not working for mysterious reasons

**Comment by ankith26** [comment]
This PR is now done from my side. I have edited the opening comment in this PR to describe the PR in more detail

**Review by cclauss — APPROVED** [review]

**Review by Starbuck5 — APPROVED** [review]
CI stuff is cool but a little over my head.

Meson change makes sense.

Code changes make sense.

After this PR does it mean people on Pi will get NEON acceleration by default?

**Comment by ankith26** [comment]
> After this PR does it mean people on Pi will get NEON acceleration by default?

Basically whichever pi has NEON support in hardware will have NEON acceleration by default (so like all pi except pi 1 and older pi zero). And whichever pi doesn't, runtime detection that we have ensures that the NEON instructions won't be invoked (which will crash with illegal instruction error). It was a long standing speculation that this wouldn't work, but now this PR has tests that pass and therefore show that this approach works.

**Comment by Starbuck5** [comment]
> Basically whichever pi has NEON support in hardware will have NEON acceleration by default (so like all pi except pi 1 and older pi zero). And whichever pi doesn't, runtime detection that we have ensures that the NEON instructions won't be invoked (which will crash with illegal instruction error). It was a long standing speculation that this wouldn't work, but now this PR has tests that pass and therefore show that this approach works.

Very nice!

This does sound like something it would be to good to verify with a tester after our next dev release though, just to be sure.

**Review by MyreMylar — APPROVED** [review]
OK, LGTM 👍

**Pull request merged**